src/EFTools/DesignXmlCore/VisualStudio/VSHelpers.cs

```diff
@@ -339,9 +340,14 @@ internal static Project GetProjectForDocument(string path, IServiceProvider serv
 
             var priority = new VSDOCUMENTPRIORITY[1];
             var isDocInProjectInt = 0;
-
+            var hr = 1;
+            
             uint foundItemId = 0;
-            var hr = vsProject.IsDocumentInProject(originalPath, out isDocInProjectInt, priority, out foundItemId);
+            ThreadHelper.JoinableTaskFactory.Run(async () => {
+                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                hr = vsProject.IsDocumentInProject(originalPath, out isDocInProjectInt, priority, out foundItemId);
+            });
+
             if (NativeMethods.Succeeded(hr) && isDocInProjectInt == 1)
             {
                 projectHierarchy = vsProject as IVsHierarchy;
```

Comment:
The rest of this method also expects the UI thread -- should we be moving the switch to the UI thread somewhere else in the stack?